commands/scanpullrequest.go

```diff
@@ -0,0 +1,224 @@
+package commands
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/jfrog/frogbot/commands/utils"
+	"github.com/jfrog/froggit-go/vcsclient"
+	audit "github.com/jfrog/jfrog-cli-core/v2/xray/commands/audit/generic"
+	xrayutils "github.com/jfrog/jfrog-cli-core/v2/xray/utils"
+	"github.com/jfrog/jfrog-client-go/utils/io/fileutils"
+	clientLog "github.com/jfrog/jfrog-client-go/utils/log"
+	"github.com/jfrog/jfrog-client-go/xray/services"
+	clitool "github.com/urfave/cli/v2"
+)
+
+func ScanPullRequest(c *clitool.Context) error {
+	params, client, err := utils.GetParamsAndClient(true)
+	if err != nil {
+		return err
+	}
+
+	if err = beforeScan(params, client); err != nil {
+		return err
+	}
+
+	return scanPullRequest(params, client)
+}
+
+// Run before scan, to make sure the Xray scan will be run only after adding the frogbot label.
+// If label is missing - create the label and do nothing
+// If pr isn't labeled - do nothing
+// If pr is labeled - remove label and allow running Xray scan (return nil)
+// params - Frogbot parameters retreived from the environment variables
+// client - The VCS client
+func beforeScan(params *utils.FrogbotParams, client vcsclient.VcsClient) error {
+	labelInfo, err := client.GetLabel(context.Background(), params.RepoOwner, params.Repo, string(utils.LabelName))
+	if err != nil {
+		return err
+	}
+	if labelInfo == nil {
+		clientLog.Info("Creating label " + string(utils.LabelName))
+		err = client.CreateLabel(context.Background(), params.RepoOwner, params.Repo, vcsclient.LabelInfo{
+			Name:        string(utils.LabelName),
+			Description: string(utils.LabelDescription),
+			Color:       string(utils.LabelColor),
+		})
+		if err != nil {
+			return err
+		}
+		return utils.ErrLabelCreated
+	}
+
+	labels, err := client.ListPullRequestLabels(context.Background(), params.RepoOwner, params.Repo, params.PullRequestID)
+	if err != nil {
+		return err
+	}
+	clientLog.Debug("The following labels found in the pull request: ", labels)
+	for _, label := range labels {
+		if label == string(utils.LabelName) {
+			clientLog.Info("Unlabeling '"+utils.LabelName+"' from pull request", params.PullRequestID)
+			err = client.UnlabelPullRequest(context.Background(), params.RepoOwner, params.Repo, string(utils.LabelName), params.PullRequestID)
```

Comment:
Maybe, it will be a better approach to unlabeled the PR after the scan? 
In that case, an error returns from client.UnlabelPullRequest will do not cancel the scan itself.